drivers/interrupt_controller/intc_exti_stm32.c

```diff
@@ -15,145 +16,71 @@
 #include <zephyr/device.h>
 #include <soc.h>
 #include <stm32_ll_bus.h>
-#include <stm32_ll_gpio.h> /* For STM32F1 series */
 #include <stm32_ll_exti.h>
 #include <stm32_ll_system.h>
+#include <zephyr/logging/log.h>
 #include <zephyr/sys/__assert.h>
 #include <zephyr/sys/util.h>
-#include <zephyr/dt-bindings/pinctrl/stm32-pinctrl-common.h> /* For STM32L0 series */
-#include <zephyr/drivers/interrupt_controller/gpio_intc_stm32.h>
+#include <zephyr/drivers/interrupt_controller/intc_exti_stm32.h>
 #include <zephyr/drivers/clock_control/stm32_clock_control.h>
-#include <zephyr/irq.h>
 
 #include "stm32_hsem.h"
+#include "intc_exti_stm32_priv.h"
 
-/** @brief EXTI lines range mapped to a single interrupt line */
-struct stm32_exti_range {
-	/** Start of the range */
-	uint8_t start;
-	/** Range length */
-	uint8_t len;
-};
+LOG_MODULE_REGISTER(exti_stm32, CONFIG_INTC_LOG_LEVEL);
 
-#define NUM_EXTI_LINES DT_PROP(DT_NODELABEL(exti), num_lines)
+bool stm32_exti_is_pending(uint32_t line)
+{
+	bool ret = false;
+	const uint32_t line_num = exti_linenum_to_ll_exti_line(line);
```

Comment:
Public APIs should accept a line number.
```suggestion
bool stm32_exti_is_pending(uint32_t line_num)
{
	bool ret = false;
	const uint32_t line = exti_linenum_to_ll_exti_line(line_num);
```
Ditto for all functions